relay-kafka/src/producer/mod.rs

```diff
@@ -77,7 +76,7 @@ pub enum ClientError {
 /// Describes the type which can be sent using kafka producer provided by this crate.
 pub trait Message {
     /// Returns the partitioning key for this kafka message determining.
-    fn key(&self) -> [u8; 16];
+    fn key(&self) -> Option<[u8; 16]>;
```

Comment:
Unrelated to the change, but I don't understand what this docstring is trying to say.